docs/sources/operations/troubleshooting/troubleshooting-ingest.md

```diff
@@ -0,0 +1,1017 @@
+
+The `reason` label on these metrics indicates the specific error type. Set up alerts on these metrics to detect ingestion problems.
+
+## Rate limit errors
+
+Rate limits protect Loki from being overwhelmed by excessive log volume. When rate limits are exceeded, Loki returns HTTP status code `429 Too Many Requests`.
+
+### Error: `rate_limited`
+
+**Error message:**
+
+`ingestion rate limit exceeded for user <tenant> (limit: <limit> bytes/sec) while attempting to ingest <lines> lines totaling <bytes> bytes, reduce log volume or contact your Loki administrator to see if the limit can be increased`
+
+**Cause:**
+
+The tenant has exceeded their configured ingestion rate limit. This is a global per-tenant limit enforced by the distributor.
+
+**Default configuration:**
+
+- `ingestion_rate_mb`: 4 MB/sec
+- `ingestion_burst_size_mb`: 6 MB
+- `ingestion_rate_strategy`: "global" (shared across all distributors)
+
+**Resolution:**
+
+* **Increase rate limits** (if you have sufficient cluster resources):
+
+   ```yaml
+   limits_config:
+     ingestion_rate_mb: 8
+     ingestion_burst_size_mb: 12
+   ```
+
+* **Reduce log volume** by:
+
+  - Filtering unnecessary log lines using Alloy processing stages
+  - Collecting logs from fewer targets
+  - Sampling high-volume streams
+  - Using Alloy's rate limiting and filtering capabilities
+
+* **Switch to local strategy** (if using single-region deployment):
+
+   ```yaml
+   limits_config:
+     ingestion_rate_strategy: local
+   ```
+
+   {{< admonition type="note" >}}
+   This multiplies the effective limit by the number of distributor replicas.
+   {{< /admonition >}}
+
+**Properties:**
+
+- Enforced by: Distributor
+- Retryable: Yes
+- HTTP status: 429 Too Many Requests
+- Configurable per tenant: Yes
+
+### Error: `stream_limit`
+
+**Error message:**
+
+`maximum active stream limit exceeded when trying to create stream <stream_labels>, reduce the number of active streams (reduce labels or reduce label values), or contact your Loki administrator to see if the limit can be increased, user: <tenant>`
+
+**Cause:**
+
+The tenant has reached the maximum number of active streams. Active streams are held in memory on ingesters, and excessive streams can cause out-of-memory errors.
+
+**Default configuration:**
+
+- `max_global_streams_per_user`: 5000 (globally)
+- `max_streams_per_user`: 0 (no local limit per ingester)
+- Active stream window: `chunk_idle_period` (default: 30 minutes)
+
+**Terminology:**
+
+- **Active stream**: A stream that has received logs within the `chunk_idle_period` (default: 30 minutes)
+
+**Resolution:**
+
+* **Reduce stream cardinality** by:
+
+  - Using fewer unique label combinations by removing unnecessary labels
+  - Avoiding high-cardinality labels (IDs, timestamps, UUIDs)
+  - Using structured metadata instead of labels for high-cardinality data
+
+* **Increase stream limits** (if you have sufficient memory):
```

Comment:
we may want a note here:

do not increase stream limits to accommodate high cardinality labels, this can result in Loki flushing extremely high numbers of small files which will make for extremely poor query performance.

as volume and the size of the infrastructure being monitored increase it would be expected to increase the stream limit, however even for hundreds of TB's a day of logs you should avoid exceeding 300,000 max global streams per user